examples/ai-core/src/e2e/togetherai.test.ts

```diff
@@ -1,369 +1,38 @@
-        messages: [
-          {
-            role: 'user',
-            content: [
-              { type: 'text', text: 'Describe the image in detail.' },
-              {
-                type: 'image',
-                image: fs
-                  .readFileSync('./data/comic-cat.png')
-                  .toString('base64'),
-              },
-            ],
-          },
-        ],
-      });
-
-      const chunks: string[] = [];
-      for await (const chunk of result.textStream) {
-        chunks.push(chunk);
-      }
-
-      const fullText = chunks.join('');
-      expect(fullText.toLowerCase()).toContain('cat');
-      expect(chunks.length).toBeGreaterThan(0);
-      expect((await result.usage)?.totalTokens).toBeGreaterThan(0);
-    });
-  });
-
-  describe.each(MODEL_VARIANTS.embedding)('Embedding Model: %s', modelId => {
-    const model = provider.textEmbeddingModel(modelId);
-
-    it('should generate single embedding', async () => {
-      const result = await embed({
-        model,
-        value: 'This is a test sentence for embedding.',
-      });
-
-      expect(Array.isArray(result.embedding)).toBe(true);
-      expect(result.embedding.length).toBeGreaterThan(0);
-      // TogetherAI API does not return usage data.
-      // expect(result.usage.tokens).toBeGreaterThan(0);
-    });
-
-    it('should generate multiple embeddings', async () => {
-      const result = await embedMany({
-        model,
-        values: [
-          'First test sentence.',
-          'Second test sentence.',
-          'Third test sentence.',
-        ],
-      });
-
-      expect(Array.isArray(result.embeddings)).toBe(true);
-      expect(result.embeddings.length).toBe(3);
-      // TogetherAI API does not return usage data.
-      // expect(result.usage.tokens).toBeGreaterThan(0);
-    });
-  });
-
-  describe.each(MODEL_VARIANTS.completion)('Completion Model: %s', modelId => {
-    const model = provider(modelId);
-
-    it('should generate text', async () => {
-      const result = await generateText({
-        model,
-        prompt: 'Complete this code: function fibonacci(n) {',
-      });
-
-      expect(result.text).toBeTruthy();
-      expect(result.usage?.totalTokens).toBeGreaterThan(0);
-    });
-
-    it('should stream text', async () => {
-      const result = streamText({
-        model,
-        prompt: 'Write a Python function that sorts a list:',
-      });
-
-      const chunks: string[] = [];
-      for await (const chunk of result.textStream) {
-        chunks.push(chunk);
-      }
-
-      expect(chunks.length).toBeGreaterThan(0);
-      expect((await result.usage)?.totalTokens).toBeGreaterThan(0);
-    });
-  });
-});
+  togetherai as provider,
+  TogetherAIErrorData,
+} from '@ai-sdk/togetherai';
+import { APICallError } from 'ai';
+import { createFeatureTestSuite } from './feature-test-suite';
+
+createFeatureTestSuite({
+  name: 'TogetherAI',
+  models: {
+    invalidModel: provider.chatModel('no-such-model'),
```

Comment:
How would this work? Wouldn't this need to be deferred?

`invalidModel: () => provider.chatModel('no-such-model')`